src/OrchardCore.Modules/OrchardCore.OpenIdConnect/Drivers/OpenIdConnectSettingsDisplayDriver.cs

```diff
@@ -0,0 +1,93 @@
+            IHtmlLocalizer<OpenIdConnectSettingsDisplayDriver> stringLocalizer,
+            IMemoryCache memoryCache,
+            IOpenIdConnectService openIdConnectService
+            )
+        {
+            _httpContextAccessor = httpContextAccessor;
+            _notifier = notifier;
+            T = stringLocalizer;
+            _memoryCache = memoryCache;
+            _openIdConnectService = openIdConnectService;
+        }
+
+        public override IDisplayResult Edit(OpenIdConnectSettings settings, BuildEditorContext context)
+        {
+            if (context.GroupId == SettingsGroupId && _memoryCache.Get(RestartPendingCacheKey) != null)
+                _notifier.Warning(T["The site needs to be restarted for the settings to take effect"]);
+
+            var requestUrl = _httpContextAccessor.HttpContext.Request.GetDisplayUrl();
+            return Shape<OpenIdConnectSettingsViewModel>("OpenIdConnectSettings_Edit", model =>
+            {
+                model.DisplayName = settings.DisplayName;
+                model.TestingModeEnabled = settings.TestingModeEnabled;
+                model.AllowedScopes = settings.AllowedScopes != null ? string.Join(",", settings.AllowedScopes) : null;
+                model.Authority = settings.Authority;
+                model.CallbackPath = settings.CallbackPath;
+                model.ClientId = settings.ClientId;
+                model.ClientSecret = _openIdConnectService.Unprotect(settings.ClientSecret);
+                model.SignedOutCallbackPath = settings.SignedOutCallbackPath;
+                model.SignedOutRedirectUri = settings.SignedOutRedirectUri;
+            }).Location("Content:2").OnGroup(SettingsGroupId);
+        }
+
+        public override async Task<IDisplayResult> UpdateAsync(OpenIdConnectSettings settings, IUpdateModel updater, string groupId)
+        {
+            if (groupId == SettingsGroupId)
+            {
+                var previousClientSecret = settings.ClientSecret;
+                var model = new OpenIdConnectSettingsViewModel();
+                await updater.TryUpdateModelAsync(model, Prefix);
+                settings.DisplayName = model.DisplayName;
+                settings.TestingModeEnabled = model.TestingModeEnabled;
+                settings.AllowedScopes = model.AllowedScopes.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
+                settings.Authority = model.Authority;
+                settings.CallbackPath = new PathString(model.CallbackPath);
+                settings.ClientId = model.ClientId;
+                settings.ClientSecret = _openIdConnectService.Protect(model.ClientSecret);
+                settings.SignedOutCallbackPath = model.SignedOutCallbackPath;
+                settings.SignedOutRedirectUri = model.SignedOutRedirectUri;
+
+                if (_openIdConnectService.IsValidOpenIdConnectSettings(settings, updater.ModelState) && _memoryCache.Get(RestartPendingCacheKey) == null)
```

Comment:
We should just `Reload` the shell, we have a good support for that now. Just add a warning in the editor that the site will be restarted for the settings to be taken into account. At some point we'll create a common text that can be reused in all the modules.

Reply:
_shellHost.ReloadShellContext(_shellSettings) assuming that the permission to manage open id client settings allows the user to reload the shell? Is there any better way to reload the shell?